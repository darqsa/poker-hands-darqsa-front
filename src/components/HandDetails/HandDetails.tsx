import { HandData } from "../../features/hands/models/Hand";

interface HandDetailsProps {
  hand: HandData;
}
const HandDetails = ({ hand }: HandDetailsProps): JSX.Element => {
  const handRoute = "../img/pokerCards/";
  const positions = ["SB", "BB", "UTG", "MP", "CO", "BTN"];

  return (
    <div className="details">
      <div className="details__hand-header">
        <div className="details__img-group">
          <img
            height={75}
            width={56.5}
            src={`${handRoute}${hand.preGame.hero.hand[0]}.webp`}
            alt={`The ${hand.preGame.hero.hand[0]} poker hand.`}
            className="details__hero-card"
          />
          <img
            height={75}
            width={56.5}
            src={`${handRoute}${hand.preGame.hero.hand[1]}.webp`}
            alt={`The ${hand.preGame.hero.hand[1]} poker hand.`}
            className="details__hero-card"
          />
        </div>
        <h2 className="details__name">{hand.handName}</h2>
      </div>
      <article className="details__info-container">
        <section className="details__game-section">
          <h3 className="details__section-heading">Game Info</h3>
          <span className="details__info">{`(${
            positions[hand.preGame.hero.position - 1]
          }) Hero - ${hand.preGame.hero.initialStack}bb`}</span>
          <span className="details__info">{`(${
            positions[hand.preGame.villains[0].position - 1]
          }) Villain - ${hand.preGame.villains[0].initialStack}bb`}</span>
        </section>
        <section className="details__game-section">
          <h3 className="details__section-heading">Preflop</h3>
          <span className="details__info">{`Action: ${hand.game.preFlop.actions[0]}`}</span>
          <span className="details__info">{`Pot: ${hand.game.preFlop.pot}`}</span>
        </section>
        {hand.game.flop && (
          <section className="details__game-section">
            <h3 className="details__section-heading">Flop</h3>
            <div className="details__board-img-group">
              <img
                height={60}
                width={45.2}
                className="hand__board-card"
                src={`${handRoute}${hand.game.flop?.board[0]}.webp`}
                alt={`The ${hand.game.flop?.board[0]} poker hand.`}
              />
              <img
                height={60}
                width={45.2}
                className="hand__board-card"
                src={`${handRoute}${hand.game.flop?.board[1]}.webp`}
                alt={`The ${hand.game.flop?.board[1]} poker hand.`}
              />
              <img
                height={60}
                width={45.2}
                className="hand__board-card"
                src={`${handRoute}${hand.game.flop.board[2]}.webp`}
                alt={`The ${hand.game.flop.board[2]} poker hand.`}
              />
            </div>
            <span className="details__info">{`Action: ${hand.game.flop.actions[0]}`}</span>
            <span className="details__info">{`Pot: ${hand.game.flop.pot}`}</span>
          </section>
        )}
        {hand.game.turn && (
          <section className="details__game-section">
            <h3 className="details__section-heading">Turn</h3>
            <div className="details__board-img-group">
              <img
                height={60}
                width={45.2}
                className="hand__board-card"
                src={`${handRoute}${hand.game.turn.board}.webp`}
                alt={`The ${hand.game.turn.board} poker hand.`}
              />
            </div>
            <span className="details__info">{`Action: ${hand.game.turn.actions[0]}`}</span>
            <span className="details__info">{`Pot: ${hand.game.turn.pot}`}</span>
          </section>
        )}
        {hand.game.river && (
          <section className="details__game-section">
            <h3 className="details__section-heading">River</h3>
            <div className="details__board-img-group">
              <img
                height={60}
                width={45.2}
                className="hand__board-card"
                src={`${handRoute}${hand.game.river.board}.webp`}
                alt={`The ${hand.game.river.board} poker hand.`}
              />
            </div>
            <span className="details__info">{`Action: ${hand.game.river.actions[0]}`}</span>
            <span className="details__info">{`Pot: ${hand.game.river.pot}`}</span>
          </section>
        )}
      </article>
      <div className="details__hand-footer">
        <span className="details__info">
          {hand.postGame.gameWinner === "hero"
            ? `(${positions[hand.preGame.hero.position - 1]}) Hero won ${
                hand.postGame.finalPot
              }bb`
            : `(${
                positions[hand.preGame.villains[0].position - 1]
              }) Villain won ${hand.postGame.finalPot}bb`}
        </span>
      </div>
    </div>
  );
};
export default HandDetails;
