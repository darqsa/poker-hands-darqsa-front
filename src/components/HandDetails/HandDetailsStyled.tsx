import styled from "styled-components";
import styles from "../../styles/styles";

const DetailsStyled = styled.div`
  display: flex;
  background-color: ${styles.colors.grey};
  flex-direction: column;
  margin: 15px 0;
  padding: 25px;
  color: ${styles.colors.main};
  border-radius: 15px;
  @media (min-width: 700px) {
    width: 65%;
    margin: 30px auto;
    padding: 25px 50px;
  }
  p {
    margin: 10px 0 0 0;
  }
  .details {
    &__info-container {
      @media (min-width: 700px) {
        display: flex;
        flex-wrap: wrap;
        gap: 30px;
      }
    }
    &__hand-header {
      display: flex;
      position: relative;
      justify-content: flex-start;
      height: 80px;
    }
    &__img-group {
      width: 80px;
    }
    &__hero-card {
      position: absolute;
      &:nth-child(2) {
        left: 22px;
        top: 16px;
      }
    }
    &__name {
      flex: 2.5;
      font-weight: ${styles.fontWeights.bolder};
      font-size: ${styles.fontSizes.big};
      font-style: italic;
      margin: 0;
      @media (min-width: 700px) {
        margin: auto 0;
      }
    }
    &__section-heading {
      margin-bottom: 0;
    }
    &__game-section {
      display: flex;
      flex-direction: column;
      gap: 5px;
      @media (min-width: 700px) {
        width: 45%;
      }
    }
    &__game-img-container {
      width: 100%;
      margin-top: 20px;
      display: flex;
      justify-content: center;
    }
    &__description-img {
      max-width: 290px;
    }
  }
`;
export default DetailsStyled;
