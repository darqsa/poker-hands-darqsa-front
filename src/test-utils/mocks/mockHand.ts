import { HandData } from "../../features/hands/models/Hand";

export const fakeHand: HandData = {
  handName: "Best hand name ever",
  preGame: {
    hero: { hand: ["Ac", "Ad"], initialStack: 100, position: 0 },
    villains: [{ hand: ["Ah", "As"], initialStack: 100, position: 1 }],
  },
  game: { preFlop: { actions: ["Everyone is allin"], pot: 200 } },
  postGame: { finalPot: 200, gameWinner: "Hero" },
  id: "1234",
};

export const completeFakeHand: HandData = {
  handName: "Best hand name ever fpodsmfpomf",
  preGame: {
    hero: { hand: ["Ac", "Ad"], initialStack: 100, position: 0 },
    villains: [{ hand: ["Ah", "As"], initialStack: 100, position: 1 }],
  },
  game: {
    preFlop: { actions: ["Everyone is allin"], pot: 200 },
    flop: {
      board: ["Ts", "9c", "8h"],
      actions: ["Everyone is allin"],
      pot: 200,
    },
    turn: { board: "7d", actions: ["Everyone is allin"], pot: 200 },
    river: { board: "6d", actions: ["Everyone is allin"], pot: 200 },
  },
  postGame: { finalPot: 200, gameWinner: "Hero" },
  id: "1234",
};
