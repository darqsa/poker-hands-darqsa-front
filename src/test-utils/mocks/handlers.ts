import { rest } from "msw";
import { fakeHand } from "./mockHand";

const apiUrl = process.env.REACT_APP_API_URL as string;

const handlers = [
  rest.post(`${apiUrl}users/register`, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json({
        username: "bobby",
        id: "6311947608ed28e35ccabbeb",
      })
    );
  }),

  rest.post(`${apiUrl}users/login`, async (req, res, ctx) => {
    return res(
      ctx.status(201),
      ctx.json({
        token:
          "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6ImFkbWluIiwiaWQiOiI2MzBkMDBjMTE1MDllNTE2N2JiN2Y1YmIiLCJpYXQiOjE2NjIwOTY0NzB9.tVQVAvCBc5CXN7-GQirdI67Q8Zuzae0EQmVBhPow11s",
      })
    );
  }),

  rest.get(`${apiUrl}hands`, async (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json({
        hands: [fakeHand],
      })
    );
  }),

  rest.post(`${apiUrl}hands/create`, async (req, res, ctx) => {
    return res(ctx.status(201), ctx.json("Hand created successfully"));
  }),

  rest.delete(`${apiUrl}hands/delete/1234`, async (req, res, ctx) => {
    return res(ctx.status(201), ctx.json("Hand deleted successfully"));
  }),
];

export default handlers;
