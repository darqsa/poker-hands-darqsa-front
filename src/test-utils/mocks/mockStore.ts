import { configureStore, createReducer } from "@reduxjs/toolkit";
import { HandData } from "../../features/hands/models/Hand";
import { LoginUserData } from "../../features/users/models/User";
import { completeFakeHand, fakeHand } from "./mockHand";

const initialState: HandData[] = [fakeHand];
const initialUiState = {
  isAlertShown: false,
  isLoadingShown: false,
  alertMessage: "",
};
const userInitialState: LoginUserData = { id: "", token: "", username: "" };

const mockHandsReducer = createReducer(initialState, (builder) => {
  builder.addDefaultCase((state) => [completeFakeHand]);
});

const mockUiReducer = createReducer(initialUiState, (builder) => {
  builder.addDefaultCase((state) => ({
    isAlertShown: true,
    isLoadingShown: true,
    alertMessage: "test message",
  }));
});

const emptyMockHandsReducer = createReducer(initialState, (builder) => {
  builder.addDefaultCase((state) => []);
});

const mockUserReducer = createReducer(userInitialState, (builder) => {
  builder.addDefaultCase((state) => state);
});

const mockStore = configureStore({
  reducer: {
    hands: mockHandsReducer,
    user: mockUserReducer,
    ui: mockUiReducer,
  },
});

export const emptyMockStore = configureStore({
  reducer: {
    hands: emptyMockHandsReducer,
    user: mockUserReducer,
    ui: mockUiReducer,
  },
});

export default mockStore;
