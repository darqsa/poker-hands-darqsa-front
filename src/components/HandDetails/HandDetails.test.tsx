import { completeFakeHand } from "../../test-utils/mocks/mockHand";
import HandDetails from "./HandDetails";
import renderer from "react-test-renderer";

describe("Given a HandDetails component", () => {
  describe("When receives a completeFakeHand by props", () => {
    test("Then it should render every property from the completeFakeHand object", () => {
      const expectedHandDetails = renderer.create(
        <HandDetails hand={completeFakeHand} />
      );

      expect(expectedHandDetails).toMatchSnapshot();
    });
  });
});
