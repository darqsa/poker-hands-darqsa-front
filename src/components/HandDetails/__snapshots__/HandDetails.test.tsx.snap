// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a HandDetails component When receives a completeFakeHand by props Then it should render every property from the completeFakeHand object 1`] = `
<div
  className="sc-bczRLJ fvPtGq details"
>
  <div
    className="details__hand-header"
  >
    <div
      className="details__img-group"
    >
      <img
        alt="The Ac poker hand."
        className="details__hero-card"
        height={60}
        src="../img/pokerCards/Ac.webp"
        width={45.2}
      />
      <img
        alt="The Ad poker hand."
        className="details__hero-card"
        height={60}
        src="../img/pokerCards/Ad.webp"
        width={45.2}
      />
    </div>
    <h2
      className="details__name"
    >
      Best hand name ever fpodsmfpomf
    </h2>
  </div>
  <article
    className="details__info-container"
  >
    <section
      className="details__game-section"
    >
      <h3
        className="details__section-heading"
      >
        Game Info
      </h3>
      <span
        className="details__info"
      >
        (SB) Hero - 100bb
      </span>
      <span
        className="details__info"
      >
        (BB) Villain - 100bb
      </span>
      <p
        className="details__info"
      >
        <strong>
          Description: 
        </strong>
        test
      </p>
    </section>
    <section
      className="details__game-section"
    >
      <h3
        className="details__section-heading"
      >
        Preflop
      </h3>
      <span
        className="details__info"
      >
        <strong>
          Action:
        </strong>
         
        Everyone is allin
      </span>
      <span
        className="details__info"
      >
        <strong>
          Pot: 
        </strong>
        200bb
      </span>
    </section>
    <section
      className="details__game-section"
    >
      <h3
        className="details__section-heading"
      >
        Flop
      </h3>
      <div
        className="details__board-img-group"
      >
        <img
          alt="The Ts poker hand."
          className="hand__board-card"
          height={60}
          src="../img/pokerCards/Ts.webp"
          width={45.2}
        />
        <img
          alt="The 9c poker hand."
          className="hand__board-card"
          height={60}
          src="../img/pokerCards/9c.webp"
          width={45.2}
        />
        <img
          alt="The 8h poker hand."
          className="hand__board-card"
          height={60}
          src="../img/pokerCards/8h.webp"
          width={45.2}
        />
      </div>
      <span
        className="details__info"
      >
        <strong>
          Action:
        </strong>
         
        Everyone is allin
      </span>
      <span
        className="details__info"
      >
        <strong>
          Pot: 
        </strong>
        200bb
      </span>
    </section>
    <section
      className="details__game-section"
    >
      <h3
        className="details__section-heading"
      >
        Turn
      </h3>
      <div
        className="details__board-img-group"
      >
        <img
          alt="The 7d poker hand."
          className="hand__board-card"
          height={60}
          src="../img/pokerCards/7d.webp"
          width={45.2}
        />
      </div>
       
      <span
        className="details__info"
      >
        <strong>
          Action:
        </strong>
         
        Everyone is allin
      </span>
      <span
        className="details__info"
      >
        <strong>
          Pot: 
        </strong>
        200bb
      </span>
    </section>
    <section
      className="details__game-section"
    >
      <h3
        className="details__section-heading"
      >
        River
      </h3>
      <div
        className="details__board-img-group"
      >
        <img
          alt="The 6d poker hand."
          className="hand__board-card"
          height={60}
          src="../img/pokerCards/6d.webp"
          width={45.2}
        />
      </div>
       
      <span
        className="details__info"
      >
        <strong>
          Action:
        </strong>
         
        Everyone is allin
      </span>
      <span
        className="details__info"
      >
        <strong>
          Pot: 
        </strong>
        200bb
      </span>
    </section>
    <section
      className="details__game-section"
    >
      <h3
        className="details__section-heading"
      >
        Summary:
      </h3>
      <span
        className="details__info"
      >
        Villain shows:
      </span>
      <div
        className="details__board-img-group"
      >
        <img
          alt="The Ah poker hand."
          className="hand__board-card"
          height={60}
          src="../img/pokerCards/Ah.webp"
          width={45.2}
        />
        <img
          alt="The As poker hand."
          className="hand__board-card"
          height={60}
          src="../img/pokerCards/As.webp"
          width={45.2}
        />
      </div>
      <span
        className="details__info"
      >
        (BB) Villain won 200bb
      </span>
    </section>
    <div
      className="details__game-img-container"
    >
      <img
        alt="Hand extra info by user"
        className="hand__description-img"
        src="https://www.ldoceonline.com/media/english/illustration/slice_pizza.jpg?version=1.2.55"
        width={290}
      />
    </div>
  </article>
</div>
`;
